tests/unittests/Foundation/ReferenceFoundation/TestNSURLResponse.mm

```diff
@@ -74,22 +77,22 @@
     ASSERT_OBJCEQ_MSG(encoding, res1.textEncodingName, @"should be the utf8 encoding");
     NSURLResponse* res2 = [[[NSURLResponse alloc] initWithURL:url MIMEType:nil expectedContentLength:0 textEncodingName:nil] autorelease];
     ASSERT_OBJCEQ(nil, res2.textEncodingName);
+    ASSERT_OBJCEQ(@"test", res1.suggestedFilename);
+    ASSERT_OBJCEQ(@"test", res2.suggestedFilename);
 }
 
-// Disabled pending GH#832; we do not respect the incoming MIME type when generating filenames. OS X does.
-OSX_DISABLED_TEST(NSURLResponse, SuggestedFilename) {
+TEST(NSURLResponse, SuggestedFilename) {
     NSURL* url = [NSURL URLWithString:@"a/test/name.extension"];
     NSURLResponse* res =
         [[[NSURLResponse alloc] initWithURL:url MIMEType:@"text/plain" expectedContentLength:0 textEncodingName:nil] autorelease];
-    ASSERT_OBJCEQ(@"name.extension", res.suggestedFilename);
```

Comment:
>name.extension [](start = 20, length = 14)

the old value was hardcoded to work with the improper implementation.
the actual values were ran on OSX and they pass.